### PR TITLE
gh-95273: Move sqlite3 executemany examples from reference to tutorial

### LINKED ISSUE #95273: Improve the sqlite3 reference
Improve the sqlite3 reference, using [Diataxis as a guideline](https://diataxis.fr/reference/). A lot of work has already been done here, with great help from Alex Waygood and CAM Gerlach. PRs so far[^1] have focused on...:

- [Be accurate](https://diataxis.fr/reference/#be-accurate)
- [Do nothing but describe](https://diataxis.fr/reference/#do-nothing-but-describe); making the wording more to the point

<details>
<summary>Previous work over the last few years</summary>

Both of these have been improved greatly by the following issues and PRs, amongst others:
- gh-94630
- gh-95235
- gh-95235
- gh-94940
- gh-94321
- gh-94628
- gh-94017
- gh-93890
- gh-93925
- gh-87260
- gh-89018
- gh-92780
- gh-90923
- gh-30660
- gh-29326
- gh-29200
- gh-29219
- gh-28238
- gh-26220
- gh-25003
- gh-23159
- gh-23965
- consistency: gh-31772
- consistency: gh-24746

</details>

There are remaining issues, though:

- [x] Be accurate:
    - #95379
    - #95523
    - #95530
- [x] [Be consistent](https://diataxis.fr/reference/#be-consistent): Most of the reference docs for sqlite3 functions and methods use the imperative mood. Some of the methods use other wordings. Suggesting to normalise, and use the imperative mood across the reference section.
  Improved with:
    - #95274 
    - #95525
    - Remove duplicate links
- [x] [Respect the structure of the machinery](https://diataxis.fr/reference/#respect-the-structure-of-the-machinery): Arrange the sections of reference to follow the architecture of the module. This is pretty good already, but I believe we may benefit from some adjustments, in particular:
    - [x] Consistently group attributes and methods
    - #95626
    - #95430
    - Align wording in function and method docstrings with documentation
- [x] [Do nothing but describe](https://diataxis.fr/reference/#do-nothing-but-describe): Avoid digressions and discussions, or even better, extract them to separate sections. Remove and or simplify examples.
    - #95351
    - #95383
    - gh-95840

[^1]: There have been too many PRs to list; some of them were done before Diataxis was even discussed.

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-95273

### CODE REVIEW

**Comment by erlend-aasland** [comment]
My 2 cents:

The executemany examples are way more complex than the average user needs. We should focus on the sqlite3 module; not on writing fancy iterators.

The only thing that bothers me, is that we do not explicitly mention that the cursor can be used as an iterator. We could add a mention in the cursor class description.

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -67,12 +67,25 @@ after restarting the Python interpreter::
...
-retrieve a single matching row, or call :meth:`~Cursor.fetchall` to get a list
-of the matching rows.
+At this point, our database only contains one row::
 
-This example uses the iterator form::
+   >>> res = cur.execute('SELECT count(rowid) FROM stocks')
+   >>> print(res.fetchone())
+   (1,)
+
+The result is a one-item :class:`tuple`;
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]:
```suggestion
The result is a one-item :class:`tuple`:
```

Seems more appropriate here?

File: Doc/library/sqlite3.rst [hunk 2]
```diff
@@ -67,12 +67,25 @@ after restarting the Python interpreter::
...
+using :meth:`~Cursor.executemany`::
+
+   >>> data = [
+       ('2006-03-28', 'BUY', 'IBM', 1000, 45.0),
+       ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0),
+       ('2006-04-06', 'SELL', 'IBM', 500, 53.0),
+   ]
+   >>> cur.executemany('INSERT INTO stocks VALUES(?, ?, ?, ?)', data)
+
+Now, retrieve the data by iterating over the result of a SELECT statement::
```
Comment (CAM-Gerlach) [inline Doc/library/sqlite3.rst]:
```suggestion
Then, retrieve the data by iterating over the result of a ``SELECT`` statement::
```

Avoid repetition of now and make ``SELECT`` a verbatim literal

**Review by CAM-Gerlach — COMMENTED** [review]
A couple minor comments, otherwise LGTM. Thanks @erlend-aasland !

**Comment by erlend-aasland** [comment]
> The only thing that bothers me, is that we do not explicitly mention that the cursor can be used as an iterator. We could add a mention in the cursor class description.

I've already got this covered in the #95273 todo list.

**Comment by AlexWaygood** [comment]
I've got quite a lot on at the moment, so I'll leave this one in @CAM-Gerlach's very capable hands, if that's okay!

**Comment by erlend-aasland** [comment]
I wonder if it may make sense to keep a minimum example in the reference, though. The two previous examples ate more than a screenful. How about leaving something like this:

```
data = [
    ("row1",),
    ("row2",),
]
cur.executemany("insert into t values(?)", data)
```

**Comment by CAM-Gerlach** [comment]
SGTM; concise examples can certainly be helpful to aiding understanding and this one does to be, at least as a novice

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks @erlend-aasland :+1:

File: Doc/library/sqlite3.rst [hunk 3]
```diff
@@ -990,12 +1003,13 @@ Cursor Objects
...
-      Here's a shorter example using a :term:`generator`:
-
-      .. literalinclude:: ../includes/sqlite3/executemany_2.py
+      Example::
 
+          data = [
+              ("row1",),
+              ("row2",),
+          ]
+          cur.executemany("insert into t values(?)", data)
```
Comment (erlend-aasland) [inline Doc/library/sqlite3.rst:1013]: Adding an explanatory comment here, as discussed in #95383:

```suggestion
          # cur is an sqlite3.Cursor object
          cur.executemany("insert into t values(?)", data)
```

**Pull request merged**